lib/metrics.js

```diff
@@ -136,26 +140,39 @@ Histogram.prototype.addPoint = function(val, timestampInMillis) {
     this.samples.push(val);
 };
 
-Histogram.prototype.flush = function() {
-    const points = [
-        this.serializeMetric(this.min, 'gauge', this.key + '.min'),
-        this.serializeMetric(this.max, 'gauge', this.key + '.max'),
-        this.serializeMetric(this.sum, 'gauge', this.key + '.sum'),
-        this.serializeMetric(this.count, 'count', this.key + '.count'),
-        this.serializeMetric(this.average(), 'gauge', this.key + '.avg')
-    ];
-
+Histogram.prototype.flush = function () {
+    let points = [];
+    if (this.aggregates.indexOf("min") !== -1) {
+      points.push(this.serializeMetric(this.min, "gauge", this.key + ".min"));
+    }
+    if (this.aggregates.indexOf("max") !== -1) {
+      points.push(this.serializeMetric(this.max, "gauge", this.key + ".max"));
+    }
+    if (this.aggregates.indexOf("sum") !== -1) {
+      points.push(this.serializeMetric(this.sum, "gauge", this.key + ".sum"));
+    }
+    if (this.aggregates.indexOf("count") !== -1) {
+      points.push(this.serializeMetric(this.count, "count", this.key + ".count"));
+    }
+    if (this.aggregates.indexOf("avg") !== -1) {
+      points.push(
+        this.serializeMetric(this.average(), "gauge", this.key + ".avg")
+      );
+    }
+  
     // Careful, calling samples.sort() will sort alphabetically giving
     // the wrong result. We must define our own compare function.
-    const numericalSortAscending = function(a, b) { return a - b; };
+    const numericalSortAscending = function (a, b) {
+      return a - b;
+    };
     this.samples.sort(numericalSortAscending);
-
-    const calcPercentile = function(p) {
-        const val = this.samples[Math.round(p * this.samples.length) - 1];
-        const suffix = '.' + Math.floor(p * 100) + 'percentile';
```

Comment:
Four space tabs are used elsewhere, prefer adherence to that style

Reply:
Yes, that makes sense too. I updated the space tabs space used in that section.